File: code/Api/utils/password.go
package utils

import (
	"golang.org/x/crypto/bcrypt"
	"superhoneypotguard/config"
)

func HashPassword(password string) (string, error) {
	cost := config.AppConfig.BCryptCost
	bytes, err := bcrypt.GenerateFromPassword([]byte(password), cost)
	return string(bytes), err
}

func ComparePassword(password, hash string) bool {
	err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
	return err == nil
}

File: code/Api/utils/password.js
const bcrypt = require('bcryptjs');
const saltRounds = parseInt(process.env.BCRYPT_ROUNDS) || 10;

const hashPassword = async (password) => {
  return await bcrypt.hash(password, saltRounds);
};

const comparePassword = async (password, hashedPassword) => {
  return await bcrypt.compare(password, hashedPassword);
};

module.exports = {
  hashPassword,
  comparePassword
};
